core/src/test/scala/org/apache/spark/scheduler/EventLoggingListenerSuite.scala

```diff
@@ -160,7 +160,7 @@ class EventLoggingListenerSuite extends FunSuite with BeforeAndAfter with Loggin
    */
   private def testApplicationEventLogging(compressionCodec: Option[String] = None) {
     val conf = getLoggingConf(testDirPath, compressionCodec)
-    val sc = new SparkContext("local", "test", conf)
+    val sc = new SparkContext("local-cluster[2,2,512]", "test", conf)
```

Comment:
Instead of making this into an integration test (the `local-cluster` mode is pretty expensive) could you modify `LocalBackend` to send an executor added event? Then you can still test with `local` mode, and also there will be less of an overall difference between the local and the cluster modes.